worlds/landstalker/Regions.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict, List, NamedTuple, Optional
+from BaseClasses import MultiWorld, Region, Entrance
+from worlds.landstalker.data.world_node import WORLD_NODES_JSON
+from worlds.landstalker.data.world_path import WORLD_PATHS_JSON
+from worlds.landstalker.data.world_region import WORLD_REGIONS_JSON
+from worlds.landstalker.data.world_teleport_tree import WORLD_TELEPORT_TREES_JSON
+
+
+class LandstalkerRegion(Region):
+    code: str
+
+    def __init__(self, code: str, name: str, player: int, multiworld: MultiWorld, hint: Optional[str] = None):
+        super().__init__(name, player, multiworld, hint)
+        self.code = code
+
+
+class LandstalkerRegionData(NamedTuple):
+    locations: Optional[List[str]]
+    region_exits: Optional[List[str]]
+
+
+def create_regions(multiworld: MultiWorld, player: int):
+    regions_table: Dict[str, LandstalkerRegion] = {}
```

Comment:
The regions table is a dictionary used to be able to fetch a region based on its "code" (internal string ID used in the Landstalker randomizer) instead of its name.
Since we are using raw data from the randomizer as .py files iin the data/ subfolder, all entities refer to that internal code instead of the region's name (i.e. locations are attached to a region using its code, etc...).

This dictionary was then built with performance in mind, to make that lookup by code faster than interating over the whole list of regions everytime to find the one with the matching code.

This could technically be changed, but would require kind of a big overhaul of the data files (which would make them less practical to update in case of a standalone randomizer update as well). I'd prefer not to do it if possible 😄